folder_service/src/main/java/com/newgen/controller/CronJobController.java

```diff
@@ -0,0 +1,162 @@
+						}
+					} else if (asyncFolderOperation.getAction() == Action.DELETE) {
+						folderService.deleteFolderAndChildren(asyncFolderOperation.getSourceFolderId(), tenantId);
+					}
+					// Update the status of the copy Folder to Completed
+					asyncFolderService.updateStatus(token, Status.COMPLETED, tenantId);
+				}
+			}
+		} catch (CustomException ex) {
+			// Change the status to FAILED if there is an exception
+			if (token != null && !token.isEmpty() && asyncFolderOperation != null) {
+				// Update the status of the async Folder Operation to Completed
+				asyncFolderService.updateStatus(token, Status.FAILED, tenantId);
+			}
+			throw ex;
+		}
+	}
+
+	public void deleteCompletedFailedAsyncFolderOperations() throws CustomException {
+		// Find all async Folder Operation which which have been completed and failed
+		// for more than 5 mins
+		List<AsyncFolderOperation> asyncFolderOperationList = asyncFolderService.findAllCompletedAndFailed();
+		for (AsyncFolderOperation asyncFolderOperation : asyncFolderOperationList) {
+			// Delete the async Folder Operation
+			String tenantId = asyncFolderOperation.getTenantId();
+			asyncFolderService.delete(asyncFolderOperation.getId(), tenantId);
+		}
+	}
+
+	public void uploadLogFile() throws CustomException, Exception, Exception {
+
+		List<File> filesInFolder = Files.walk(Paths.get("d:/log")).filter(Files::isRegularFile).map(Path::toFile)
+				.collect(Collectors.toList());
+
+		CloudBlobContainer blobcontainer = null;
+		CloudStorageAccount storageAccount = null;
+		CloudBlockBlob blob = null;
+		CloudBlobClient blobClient = null;
+		String conn = "DefaultEndpointsProtocol=https;AccountName=csg0dd2183fd41dx444bx992;AccountKey=KAiydalpISNtxMXW/7+RhpKnRS89u4JHT6mihm24pUfU1ah+dPIAvGutKrzkDR+XyNC/AzFc3v8CCfMLUx82fQ==;EndpointSuffix=core.windows.net";
+
+		storageAccount = CloudStorageAccount.parse(conn);
+		blobClient = storageAccount.createCloudBlobClient();
+		blobcontainer = blobClient.getContainerReference("logs");
+		blobcontainer.createIfNotExist();
+
+		for (int i = 0; i < filesInFolder.size(); i++) {
+			String filename = filesInFolder.get(i).getName();
+			String path = filesInFolder.get(i).getPath();
+
+			String deleteDate = deleteRecords();
+			blob = blobcontainer.getBlockBlobReference(deleteDate + filename);
+			// boolean status = blob.deleteIfExists();
+
+			String localDate = getLocalDate();
+			blob = blobcontainer.getBlockBlobReference(localDate + filename);
+			File sourceFile = new File(path);
+			FileInputStream inputStream = new FileInputStream(sourceFile);
+			blob.upload(inputStream, sourceFile.length());
+		}
+	}
+
+	private String getLocalDate() {
+		SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy-HH");
+		Date date = new Date();
+		String localDate = formatter.format(date);
+		localDate = "logs-" + localDate + "/";
+		return localDate;
+	}
+
+	private String deleteRecords() {
+		SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy-HH");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you don’t call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.